File: Java/DesignPatterns/src/main/java/designPatterns/behavioral/state/MainState.java
package designPatterns.behavioral.state;

public class MainState {
  public static void main(String[] args) {
    Context context = new Context();

    State startState = new ConcreteState.StartState();
    System.out.println(startState.doAction(context) + " : " + context.getState());

    State stopState = new ConcreteState.StopState();
    System.out.println(stopState.doAction(context) + " : " + context.getState());
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/behavioral/state/MainState.kt
package behavioral.state

fun main() {
  var context = Context()

  var startState = ConcreteState.StartState()
  println("${startState.doAction(context)} : ${context.state}")

  var stopState = ConcreteState.StopState()
  println("${stopState.doAction(context)} : ${context.state}")
}